components/list/list-item-expand-collapse-mixin.js

```diff
@@ -0,0 +1,161 @@
+				padding: 0.4rem 0 0 0.3rem;
+			}
+			.d2l-list-expand-collapse d2l-button-icon {
+				--d2l-button-icon-min-height: 1.2rem;
+				--d2l-button-icon-min-width: 1.2rem;
+			}
+			.d2l-list-expand-collapse:hover d2l-button-icon {
+				background-color: var(--d2l-button-icon-background-color-hover);
+				border-radius: var(--d2l-button-icon-border-radius);
+			}
+			:host([_render-expand-collapse-slot]) .d2l-list-expand-collapse {
+				padding: 0.4rem 0.3rem 0 0;
+				width: 1.2rem;
+			}
+			.d2l-list-expand-collapse-action {
+				cursor: pointer;
+				display: block;
+				height: 100%;
+			}
+			.d2l-list-children-loading {
+				display: flex;
+				justify-content: center;
+				padding: 0.4rem;
+			}
+		` ];
+
+		super.styles && styles.unshift(super.styles);
+		return styles;
+	}
+
+	constructor() {
+		super();
+		this._siblingHasNestedItems = false;
+		this._renderExpandCollapseSlot = false;
+		this._showChildrenLoadingSpinner = false;
+		this._parentChildUpdateSubscription = new EventSubscriberController(this, {}, { eventName: 'd2l-list-child-status' });
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		// mixin requires key for events
+		if (!this.key && this.expandable) {
+			console.warn('ListItemExpandCollapseMixin requires a key.');
+			this.expandable = false;
+		}
+	}
+
+	updated(changedProperties) {
+		if (changedProperties.has('_siblingHasNestedItems') || changedProperties.has('expandable')) {
+			this._renderExpandCollapseSlot = this.expandable || this._siblingHasNestedItems;
+		}
+		if (changedProperties.has('_draggingOver') && this._draggingOver && this.dropNested && !this.expanded && this.expandable) {
+			let elapsedHoverTime = 0;
+			let dragIntervalId = null;
+			const watchDraggingOver = () => {
+				if (elapsedHoverTime === dragHoverDropTime) {
+					if (this._draggingOver) {
+						this._toggleExpandCollapse();
+					}
+					clearInterval(dragIntervalId);
+				} else {
+					if (!this._draggingOver) {
+						clearInterval(dragIntervalId);
+					} else {
+						elapsedHoverTime += dragIntervalDelay;
+					}
+				}
+			};
+			// check if they are still hovered over same item every 100ms
+			dragIntervalId = setInterval(watchDraggingOver, dragIntervalDelay);
+		}
+		if (changedProperties.has('expanded') || changedProperties.has('_hasChildren')) {
+			this._showChildrenLoadingSpinner = this.expanded && !this._hasChildren;
+		}
+	}
+
+	updateSiblingHasChildren(siblingHasNestedItems) {
+		this._siblingHasNestedItems = siblingHasNestedItems;
+	}
+
+	_renderChildrenLoadingSpinner() {
+		if (!this.expandable || !this._showChildrenLoadingSpinner) {
+			return nothing;
+		}
+		return html`
+			<div class="d2l-list-children-loading">
+				<d2l-loading-spinner size="40"></d2l-loading-spinner>
+			</div>`;
+	}
+
+	_renderExpandCollapse() {
+		return html`
+			<div slot="expand-collapse" class="d2l-list-expand-collapse" @click="${this._toggleExpandCollapse}">
+				${this.expandable ? html`<d2l-button-icon
```

Comment:
Maybe this should be (like checkbox):
```javascript
return this.expandable ? html`
	<d2l-button-icon ...></d2l-button-icon>
` : nothing;
```
And the `<div slot="expand-collapse">...</div>` be moved out to the `ListItemMixin` where it controls the layout (what slot of the generic layout this renders into), like the rest of the slots. I think originally, one of the ideas with these lower level mixins was to make them agnostic to the actual layout or slots being used. That may or may not have been done effectively, but the `ListItemMixin` is the thing that tries to integrate these things into the layout.

Reply:
I could move the `d2l-list-expand-collapse` div to the parent `ListItemMixin` but we would want to move all the corresponding styles as well (which is the majority of them in this mixin). For the sake of code cleanliness/readability I would prefer to avoid this move. Willing to move if there is a strong preference though.

Reply:
My preference is to centralize all the layout in two places - `ListItemMixin` and `d2l-list-item-generic-layout` so that we know where all the layout is managed. But it's not a huge deal... a lot of this stuff is already commingled, and my intention is that these lower level mixins are never used anywhere except `ListItemMixin`.

Reply:
I moved the slot div to `ListItemMixin` but kept the specific styles in the `ListItemExpandCollapseMixin`... not ideal but don't really want all the styles getting lost in the larger parent mixin.